proposals/0400-init-accessors.md

```diff
@@ -358,7 +384,71 @@ init(x: Int, y: Int) {
 }
 ```
 
-Use cases for `init` accessors that provide a projection of a stored property as various units through several computed properties don't have a single preferred unit from which to initialize. Most likely, these use cases want a different member-wise initializer for each unit that you can initialize from. If a type contains several computed properties with `init` accessors that initialize the same stored property, a member-wise initializer will not be synthesized.
+Therefore, the compiler will order the initializations in the synthesized memberwise initializer to respect the `accesses` clauses:
+
+```swift
+init(x: Int, y: Int) {
+  self.y = y
+  self.x = x
+}
+```
+
+The initial review of this proposal suppressed the memberwise initializer in such cases, based on a concern that out-of-order initialization would cause surprises. However, given the fact that the fields are initialized independently (or have `accessses` relationships that defiee their relative ordering), and that side effects here are limited to those of the `init` accessors themselves, one has to introduce global side effects during initialization to observe any difference.
+
+There remain cases where a memberwise initializer cannot be synthesized. For example, if a type contains several computed properties with `init` accessors that initialize the same stored property, it is not clear which computed property should be used within the member-wise initializer. In such cases, a member-wise initializer will not be synthesized.
+
+### Init accessors on computed properties
+
+An init accessor can be provided on a computed property, in which case it is used for initialization and as a default argument in the memberwise initializer. For example, given the following:
```

Comment:
I think we should specify the behavior when a computed property with an `init` accessor has an initial value and the type has an explicit initializer:

```swift
struct S {
  var _x: Int

  var x: Int = 0 {
    @storageRestrictions(initializes: degrees)
    init(initialValue) {
      degrees = initialValue * 180 / .pi
    }

    get { ... }
    set { ... }
  }

  var y: Int

  init() {
    self.y = 10
  }
}
```

Either this is an error, or the `init` accessor for `x` is invoked with `0` before `self.y = 10`

Reply:
Great point, added in https://github.com/apple/swift-evolution/pull/2096/commits/2daa53fe319573171e6de74b007413c76b99efed